File: JavaScript/models/Cordyceps.js
// Abstracción: clase base que define comportamiento común
class Cordyceps {
    constructor(origin) {
        // Encapsulamiento
        this._origin = origin || "Desconocido";
    }

    // Método sin retorno
    spread() {
        console.log("El hongo se propaga desde: " + this._origin);
    }

    // Método con retorno (polimorfismo)
    behavior() {
        throw new Error("Este método debe ser implementado por la subclase.");
    }
}

export default Cordyceps;

File: GO/models/cordyceps.go
package models

import "fmt"

// Interfaz Cordyceps (es como una clase abstracta)
type Cordyceps interface {
    Spread()            // Método sin retorno
    Behavior() string   // Método con retorno
}

// "herencia" (esto es una composicion)
type BaseCordyceps struct {
    origin string // Atributo privado simulado
}

// metodo comun sin retorno
func (b *BaseCordyceps) Spread() {
    fmt.Println("El hongo se propaga desde: " + b.origin)
}
